app.py

```diff
@@ -63,56 +67,80 @@ def about():
 def send_js(path):
     return send_from_directory('scripts', path)
 
+@app.route('/images/<path:path>')
+def send_img(path):
+    return send_from_directory('images', path)
+
 @app.route('/temp/<path:path>')
 def send_temp(path):
     return send_from_directory('temp', path)
 
-def delete_temp_contents():
-    files = glob.glob('temp/*')
-    for f in files:
-        os.remove(f)
-
 @app.route('/upload', methods=['GET', 'POST'])
 def upload():
     if request.method == 'POST':
+        # Save audio file from POST data
         f = request.files['audio_data']
         f.save('temp/upload.wav')
+
+        # Re-encode audio file ready for transcription
         data, samplerate = sf.read('temp/upload.wav')
         sf.write('temp/upload2.wav', data, samplerate, subtype='PCM_16')
 
+        # Transcribe audio file
         with open('temp/upload2.wav', 'rb') as file:
-            print(file)
-            text = asyncio.run(transcriber.transcribe_audio(file))
-        print(text)
-        language = lang.tts_langs()[session["language-code"][:2]]
-        translated_t = chatbot.translate(text, language, "English")
-        answer = chatbot.chat(translated_t)
-        translated_a = chatbot.translate(answer, "English", language)
-        tts = gTTS(translated_a, lang=session["language-code"][:2])
-        delete_temp_contents()
-        curr_time = str(round(time.time()))
-        tts.save("temp/tts"+curr_time+".wav")
-        response_dict = {'user': text, 'ai': translated_a, 'time': curr_time}
+            text = asyncio.run(transcriber.transcribe_audio(file, session.get("language-code", "en-GB")))
+
+        # Get response from ChatBot and generate speech from it
+        answer = chat(text)
+        tts_file = tts(answer, session["language-code"])
+
+        response_dict = {'user': text, 'ai': answer, 'tts_file': tts_file}
         return json.dumps(response_dict)
     return "fail"
 
 @app.route('/initial_prompt', methods=['GET', 'POST'])
 def initial_prompt():
     if request.method == 'POST':
-        prompt = "What is your opinion of "+app_data["topic"]+"?"
-        chatbot.prompt += f"\nFriend_q: " + prompt
-        language = lang.tts_langs()[session["language-code"][:2]]
-        translated_p = chatbot.translate(prompt, "English", language)
-        response_dict = {'ai': translated_p}
+        # Start with a basic question as the initial prompt to the user, append this to the GPT-3 hidden prompt
+        start_prompt = f"What is your opinion of {session['subtopic']}?"
+        session["prompt"] += f"\nFriend_q: " + start_prompt

```

Comment:
Three things:
- Nitpick, but initial prompt is clunky for some topics or subtopics e.g. "What is your opinion of [environment]?"
- FYI initial prompt is also still broken (but only sometimes)
- You might enjoy this prompt Simon just gave me on the topic of environment: "I think extinction is a natural process that will happen to all living species."

Reply:
Initial prompt now works slightly more reliably